src/simulations/modeling/driven.py

```diff
@@ -165,3 +160,144 @@ def adjust_fluxes2model(
         model.objective = prob.Objective(new_obj, direction="min")
         solution = model.optimize(raise_error=True)
     return solution
+
+
+def flexibilize_proteomics(model, biomass_reaction, growth_rate, proteomics):
+    """
+    Replace proteomics measurements with a set that enables the model to grow. Proteins
+    are removed from the set iteratively based on sensitivity analysis (shadow prices).
+
+    Parameters
+    ----------
+    model: cobra.Model
+        The enzyme-constrained model.
+    biomass_reaction: str
+        The id of the biomass reaction in the given model.
+    proteomics: list(dict)
+        List of measurements matching the `Proteomics` schema.
+    growth_rate: dict
+        Growth rate, matching the `GrowthRate` schema.
+
+    Returns
+    -------
+    growth_rate: dict
+        New growth rate (will change if the model couldn't grow at the inputted value).
+    proteomics: list(dict)
+        Filtered list of proteomics.
+    """
+
+    # reset growth rate in model:
+    model.reactions.get_by_id(biomass_reaction).bounds = (0, 1000)
+
+    # build a table with protein ids, met ids in model and values to constrain with:
+    prot_df = pd.DataFrame()
+    for protein in proteomics:
+        protein_id = protein["identifier"]
+        lb, ub = bounds(protein["measurement"], protein["uncertainty"])
+        for met in model.metabolites:
+            if protein_id in met.id:
+                new_row = pd.DataFrame(
+                    data={"met_id": met.id, "value": ub}, index=[protein_id]
+                )
+                prot_df = prot_df.append(new_row)
```

Comment:
Alternative suggestion using cobrapy's `query` (it might make some optimizations, I'm not sure though):

```python
for met in model.metabolites.query(lambda m: protein_id in m.id):
    new_row = pd.DataFrame(
        data={"met_id": met.id, "value": ub}, index=[protein_id]
    )
    prot_df = prot_df.append(new_row)
```

Reply:
done in commit [`1fe6f9a`](https://github.com/DD-DeCaF/simulations/pull/147/commits/1fe6f9ad49497691b7f7feea0d7a10233907d456) (doesn't seem to decrease performance)